docs/src/main/sphinx/security/inspect-pem.rst

```diff
@@ -0,0 +1,139 @@
+
+Follow the steps in this topic to inspect and validate PEM-encoded key and
+certificate files. See :ref:`troubleshooting_keystore` to validate JKS keystores.
+
+.. _inspect_pems:
+
+Inspect PEM key files
+---------------------
+
+You may receive a single file that includes a private key and its certificate,
+or separate files. The file name extensions shown on this page are examples
+only; there is no extension naming standard.
+
+First, use the ``cat`` command to view these plain text files. For example:
+
+.. code-block:: shell
+
+    cat clustercoord.key
+    cat clustercoord.cert
+
+or:
+
+.. code-block:: shell
+
+    cat clustercoord.pem
+
+A key file looks something like the following:
+
+.. code-block:: text
+
+  -----BEGIN PRIVATE KEY-----
+  MIIEowIBAAKCAQEAwJL8CLeDFAHhZe3QOOF1vWt4Vuk9vyO38Y1y9SgBfB02b2jW
+  ....
+  -----END PRIVATE KEY-----
+
+If your key file reports ``BEGIN ENCRYPTED PRIVATE KEY`` instead, this means the
+key is encrypted and you must use the password to open or inspect the key. You
+may have specified the password when requesting the key, or the password could
+be assigned by your site's network managers.
+
+  If your key file reports ``BEGIN EC PRIVATE KEY`` or
+  ``BEGIN DSA PRIVATE KEY``, this designates keys using Elliptical Curve or
+  DSA alternatives to RSA. In these cases, consult ``openssl`` references for
+  the appropriate versions of the verification commands below.
+
+Inspect PEM certificate files
+-----------------------------
+
+A separate certificate file looks like the following example with the ``cat``
+command:
+
+.. code-block:: text
+
+  -----BEGIN CERTIFICATE-----
+  MIIDujCCAqICAQEwDQYJKoZIhvcNAQEFBQAwgaIxCzAJBgNVBAYTAlVTMRYwFAYD
+  ....
+  -----END CERTIFICATE-----
+  -----BEGIN CERTIFICATE-----
+  MIIDwjCCAqoCCQCxyqwZ9GK50jANBgkqhkiG9w0BAQsFADCBojELMAkGA1UEBhMC
+  ....
+  -----END CERTIFICATE-----
+
+The file can show a single certificate section, or more than one to express a
+chain of authorities, each certifying the previous.
+
+If you received a single file, make sure it shows at least one ``KEY`` and one
+``CERTIFICATE`` section. If you received separate files, concatenate them into
+one, typically in order from key to certificate. For example:
+
+.. code-block:: shell
+
+  cat clustercoord.key clustercoord.cert > clustercoord.pem
+
+.. _validate_pems:
+
+Validate PEM keys
+-----------------
+
+This page presumes your system provides the ``openssl`` command from OpenSSL 1.1
+or later.
+
+Test an RSA private key's validity with the following command:
+
+.. code-block:: text
+
+  openssl rsa -in clustercoord.pem -check -noout
+
+Look for the following confirmation message:
+
+.. code-block:: text
+
+  RSA key ok
+
+Validate PEM certificates
+-------------------------
+
+Analyze a certificate file, or the certificate portion of a concatenated file,
+with the following ``openssl`` command:
+
+.. code-block:: text
+
+  openssl x509 -in server.pem -text -noout
```

Comment:
This name is inconsistent with the examples above... change to `clustercoord.pem`

Reply:
I missed that one. Fixed now, thanks.